src/Network/Response.php

```diff
@@ -677,10 +739,50 @@ public function location($url = null)
     public function body($content = null)
     {
         if ($content === null) {
-            return $this->_body;
+            $this->stream->rewind();
+            $result = $this->stream->getContents();
+            if (empty($result)) {
+                return null;
+            }
+
+            return $result;
         }
 
-        return $this->_body = $content;
+        // BC compatibility
+        if (is_callable($content)) {
+            $content = $this->_handleCallableBody($content);
```

Comment:
The CakePHP one is backwards compatible with how body() works now (no string is expected to be returned out of the callback)

The current `body` method can't be expected to work in an immutable context as it has to be backwards compatible. I think once this set of changes merge, we'll do a round of deprecations on methods that overlap the PSR7 interface and add in any missing get/with methods.